src/coreclr/nativeaot/System.Private.TypeLoader/src/Internal/Runtime/TypeLoader/CallConverterThunk.cs

```diff
@@ -1119,13 +1119,11 @@ private static unsafe void InvokeTarget(void* allocatedStackBuffer, ref CallConv
                         RuntimeTypeHandle returnTypeRuntimeTypeHandle = thRetType.GetRuntimeTypeHandle();
 
                         // Need to box value type before returning it
-                        object returnValue;
+                        object returnValue = null;
                         if (returnType == CorElementType.ELEMENT_TYPE_BYREF && returnValueToCopy == null)
                         {
                             // This is a byref return and dereferencing it would result in a NullReferenceException.
-                            // Set the return value to a sentinel that InvokeUtils will recognize.
-                            // Can't throw from here or we would wrap this in a TargetInvocationException.
-                            returnValue = InvokeUtils.NullByRefValueSentinel;
+                            CompilerHelpers.ThrowHelpers.ThrowInvokeNullRefReturned();
```

Comment:
This way the `object returnValue = null;` above is not needed and the compiler will keep warning us if we ever add a branch that forgot to initialize the value.

```suggestion
                            CompilerHelpers.ThrowHelpers.ThrowInvokeNullRefReturned();
                            returnValue = null; // unreachable
```